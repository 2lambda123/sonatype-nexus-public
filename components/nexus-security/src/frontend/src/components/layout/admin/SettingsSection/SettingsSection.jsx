/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
import PropTypes from 'prop-types';
import React, {useEffect, useRef, useState} from 'react';

import './SettingsSection.scss';

import UIStrings from '../../../../constants/UIStrings';

export default function SettingsSection({isLoading, children}) {
  const wrapperRef = useRef();
  const [loadingMaskHeight, setLoadingMaskHeight] = useState(0);

  useEffect(() => {
    const wrapperHeight = wrapperRef.current.clientHeight;
    if (wrapperHeight !== loadingMaskHeight) {
      setLoadingMaskHeight(wrapperHeight);
    }
  });

  let loadingMask = null;
  if (isLoading) {
    const loadingMaskStyle = {
      height: loadingMaskHeight
    };
    loadingMask = <span style={loadingMaskStyle} className='nxrm-settings-section-loading-mask'>
      {UIStrings.SETTINGS.LOADING_MASK}
    </span>;
  }

  return <>
    {loadingMask}
    <div className='nxrm-settings-section' ref={wrapperRef}>
      {children}
    </div>
  </>;
}

SettingsSection.propTypes = {
  isLoading: PropTypes.bool
};

SettingsSection.FieldWrapper = function({labelText, children}) {
  const WrapperElement = labelText ? 'label' : 'div';

  return <WrapperElement className='nxrm-settings-section-field-wrapper'>
    {labelText ? <span className='nxrm-settings-section-field-wrapper-label'>{labelText}</span> : null}
    {children}
  </WrapperElement>;
};

SettingsSection.FieldWrapper.propTypes = {
  labelText: PropTypes.string
};

SettingsSection.Footer = function({children}) {
  return <div className='nxrm-settings-section-footer'>
    {children}
  </div>;
};
