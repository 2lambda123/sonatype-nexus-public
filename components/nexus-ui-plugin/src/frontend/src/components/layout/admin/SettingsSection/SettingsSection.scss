/**
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
/**
 * @since 3.21
 */
@import '../../../../styles/colors';

.nxrm-settings-section {
  background-color: $white;
  border: solid 1px $light-gray;
  padding: 12px;
  margin-bottom: 1em;
}

.nxrm-content-body > .nxrm-settings-section:last-child {
  margin-bottom: 0;
}

.nxrm-settings-section-loading-mask {
  align-items: center;
  background-color: $transparent;
  color: $white;
  display: flex;
  font-size: 1.5em;
  justify-content: center;
  position: absolute;
}

.nxrm-settings-section-field-wrapper {
  display: block;
  margin-bottom: 15px;

  .nxrm-settings-section-field-wrapper-label {
    display: block;
    font-weight: bold;
    margin-bottom: 4px;
  }
  .nxrm-settings-section-field-wrapper-description {
    display: block;
    margin-bottom: 2px;
    font-size: 10px;
  }
}

.nxrm-settings-section-footer {
  display: flex;
  margin-top: 10px;

  & > .nx-btn {
    margin-right: 10px;
  }
}
