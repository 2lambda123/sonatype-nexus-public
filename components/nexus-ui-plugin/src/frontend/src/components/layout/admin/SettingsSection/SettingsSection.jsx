/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
import PropTypes from 'prop-types';
import React, {createRef} from 'react';

import './SettingsSection.scss';

import UIStrings from '../../../../constants/UIStrings';
import FieldWrapper from "../../../widgets/FieldWrapper/FieldWrapper";

/**
 * @since 3.21
 */
export default class SettingsSection extends React.Component {
  state = {
    loadingMaskSize: {
      height: 0,
      width: 0
    }
  };

  settingsSectionRef = createRef();

  updateLoadingMaskSize = () => {
    if (this.settingsSectionRef.current) {
      const {height, width} = this.settingsSectionRef.current.getBoundingClientRect();
      this.setState({
        loadingMaskSize: {
          height: height,
          width: width
        }
      });
    }
  };

  componentDidMount() {
    window.addEventListener('resize', this.updateLoadingMaskSize);
    this.updateLoadingMaskSize();
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.updateLoadingMaskSize);
  }

  render() {
    const loadingMask = <span style={this.state.loadingMaskSize} className="nxrm-settings-section-loading-mask">
      {UIStrings.SETTINGS.LOADING_MASK}
    </span>;

    return <>
      {this.props.isLoading && loadingMask}
      <div className='nxrm-settings-section' ref={this.settingsSectionRef}>
        {this.props.children}
      </div>
    </>;
  }
}

SettingsSection.propTypes = {
  isLoading: PropTypes.bool
};

SettingsSection.FieldWrapper = FieldWrapper;

SettingsSection.Footer = function({children}) {
  return <div className='nxrm-settings-section-footer'>
    {children}
  </div>;
};
