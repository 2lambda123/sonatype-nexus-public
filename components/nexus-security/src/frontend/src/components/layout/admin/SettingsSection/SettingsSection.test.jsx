/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
import {shallow} from 'enzyme';
import React from 'react';

import SettingsSection from './SettingsSection';

describe('SettingsSection', () => {
  it('renders correctly', () => {
    expect(shallow(
        <SettingsSection>
          <span>test</span>
        </SettingsSection>
    )).toMatchSnapshot();
  });

  it('renders correctly while isLoading', () => {
    expect(shallow(
        <SettingsSection isLoading={true}>
          <span>test</span>
        </SettingsSection>
    )).toMatchSnapshot();
  });

  describe('SettingsSection.FieldWrapper', () => {
    it('renders correctly without label', () => {
      expect(shallow(
          <SettingsSection.FieldWrapper>
            <input type='text'/>
          </SettingsSection.FieldWrapper>
      )).toMatchSnapshot();
    });

    it('renders correctly with label', () => {
      expect(shallow(
          <SettingsSection.FieldWrapper labelText='test label'>
            <input type='text'/>
          </SettingsSection.FieldWrapper>
      )).toMatchSnapshot();
    });
  });

  describe('SettingsSection.Footer', () => {
    it('renders correctly', () => {
      expect(shallow(
          <SettingsSection.Footer>
            <button>test</button>
          </SettingsSection.Footer>
      )).toMatchSnapshot();
    });
  });
});
