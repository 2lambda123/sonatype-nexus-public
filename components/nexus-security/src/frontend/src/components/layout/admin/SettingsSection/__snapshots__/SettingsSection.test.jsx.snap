// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsSection SettingsSection.FieldWrapper renders correctly with label 1`] = `
<label
  className="nxrm-settings-section-field-wrapper"
>
  <span
    className="nxrm-settings-section-field-wrapper-label"
  >
    test label
  </span>
  <input
    type="text"
  />
</label>
`;

exports[`SettingsSection SettingsSection.FieldWrapper renders correctly without label 1`] = `
<div
  className="nxrm-settings-section-field-wrapper"
>
  <input
    type="text"
  />
</div>
`;

exports[`SettingsSection SettingsSection.Footer renders correctly 1`] = `
<div
  className="nxrm-settings-section-footer"
>
  <button>
    test
  </button>
</div>
`;

exports[`SettingsSection renders correctly 1`] = `
<Fragment>
  <div
    className="nxrm-settings-section"
  >
    <span>
      test
    </span>
  </div>
</Fragment>
`;

exports[`SettingsSection renders correctly while isLoading 1`] = `
<Fragment>
  <span
    className="nxrm-settings-section-loading-mask"
    style={
      Object {
        "height": 0,
      }
    }
  >
    Loading...
  </span>
  <div
    className="nxrm-settings-section"
  >
    <span>
      test
    </span>
  </div>
</Fragment>
`;
